ipinfo/cmd_download.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/ipinfo/cli/lib/complete"
+	"github.com/ipinfo/cli/lib/complete/predict"
+	"github.com/spf13/pflag"
+)
+
+const dbDownloadURL = "https://ipinfo.io/data/free/"
+
+var completionsDownload = &complete.Command{
+	Flags: map[string]complete.Predictor{
+		"--asn":         predict.Nothing,
+		"--country":     predict.Nothing,
+		"--country-asn": predict.Nothing,
+		"-f":            predict.Nothing,
+		"--format":      predict.Nothing,
+		"-t":            predict.Nothing,
+		"--token":       predict.Nothing,
+		"-h":            predict.Nothing,
+		"--help":        predict.Nothing,
+	},
+}
+
+func printHelpDownload() {
+	fmt.Printf(
+		`Usage: %s download [<opts>]
+
+Description:
+    Download the free ipinfo databases.
+
+Examples:
+    # Download country database in csv format.
+    $ %[1]s download --country --csv
+
+Options:
+  General:
+    --token <tok>, -t <tok>
+      use <tok> as API token.
+    --asn
+      download the free ipinfo asn database. 
+    --country
+      download the free ipinfo country database. 
+    --country-asn
+      download the free ipinfo country asn database. 
```

Comment:
These don't make sense as flags; they should be positional arguments. eg.

```
ipinfo download asn
ipinfo download country
ipinfo download country_asn
```

If the user makes a typo or the argument is not recognized, we can throw an error and show the correct types. See for example the error message in `ipinfo 1.1.1.1 --field qwieqiwe`

Reply:
The reason for going for the flag approach was it'll allow users to download multiple files at the same time. If we go with the argument approach we will only be allowing one file per download command and considering we want the 3rd argument optional as the file name then we can't really allow multiple file downloads so should we restrict the user then to only one file?

Reply:
In your implementation it's already doing it sequentially so doesn't make much difference in terms of speed. In reality, most people will only download 1 file, likely `country_asn`.